mapeditor/resourceExtractor/ResourceConverter.cpp

```diff
@@ -0,0 +1,91 @@
+#include "StdInc.h"
+
+#include "ResourceConverter.h"
+
+#include "../lib/JsonNode.h"
+#include "../lib/VCMIDirs.h"
+#include "../lib/filesystem/Filesystem.h"
+
+#include "BitmapHandler.h"
+#include "Animation.h"
+
+#include "boost/filesystem/path.hpp"
+#include "boost/locale.hpp"
+
+namespace bfs = boost::filesystem;
+
+// converts all pcx files from /Images into PNG
+void ConvertPcxToPng(bool deleteOriginals)
+{
+	bfs::path imagesPath = VCMIDirs::get().userCachePath() / "extracted" / "Images";
+	bfs::directory_iterator end_iter;
+
+	for ( bfs::directory_iterator dir_itr(imagesPath); dir_itr != end_iter; ++dir_itr )
+	{
+		try
+		{
+			if (!bfs::is_regular_file(dir_itr->status()))
+				return;
+
+			std::string filePath = dir_itr->path().string();
+			std::string fileStem = dir_itr->path().stem().string();
+			std::string filename = dir_itr->path().filename().string();
+			filename = boost::locale::to_lower(filename);
+
+			if(filename.find(".pcx") != std::string::npos)
+			{
+				auto img = BitmapHandler::loadBitmap(filename);
+				bfs::path pngFilePath = imagesPath / (fileStem + ".png");
+				img.save(QStringFromPath(pngFilePath), "PNG");
+
+				if (deleteOriginals)
+					bfs::remove(filePath);
+			}
+		}
+		catch ( const std::exception & ex )
+		{
+			logGlobal->info(dir_itr->path().filename().string() + " " + ex.what() + "\n");
+		}
+	}
+}
+
+// splits a def file into individual parts and converts the output to PNG format
+void SplitDefFile(std::string fileName, bfs::path spritesPath, bool deleteOriginals)
+{
+	if (CResourceHandler::get()->existsResource(ResourceID("SPRITES/" + fileName)))
```

Comment:
there is no guarantee that spritesPath is same path as ResourceID("SPRITES/"  +...)
can you work either with Resources or with filesystem directly?

Reply:
I try to split a def file from extracted folder... nevertheless I want to test that the resource exists so I can create an Animation object.
That is why I mix the both... and since I started from Original files they must be the same.
Or... can you suggest more concretely another way?